#! /usr/bin/env python

# 双目摄像头合为一个video设备

PACKAGE='cameras'

from dynamic_reconfigure.parameter_generator_catkin import *

RECONFIGURE_CLOSE = 3  # Parameters that need a sensor to be stopped completely when changed
RECONFIGURE_STOP = 1  # Parameters that need a sensor to stop streaming when changed
RECONFIGURE_RUNNING = 0 # Parameters that can be changed while a sensor is streaming

gen = ParameterGenerator()

gen.add("camera_index", int_t, RECONFIGURE_CLOSE,
        "Left camera device index, e.g. 0 for /dev/video0", 0, 0, 10)

gen.add("frame_width", int_t, RECONFIGURE_CLOSE,
        "Frame width", 640, 1, 2000)

gen.add("frame_height", int_t, RECONFIGURE_CLOSE,
        "Frame height", 480, 1, 2000)

gen.add("fps", double_t, RECONFIGURE_RUNNING,
        "Frames Per Second (FPS)", 30., 0., 120.)

gen.add("fourcc", str_t, RECONFIGURE_CLOSE,
        "FourCC code", "YUYV")

gen.add("brightness", double_t, RECONFIGURE_RUNNING,
        "Brightness", 0.5, 0., 1.)

gen.add("contrast", double_t, RECONFIGURE_RUNNING,
        "Contrast", 1., 0., 1.)

gen.add("saturation", double_t, RECONFIGURE_RUNNING,
        "Saturation", 1., 0., 1.)

gen.add("hue", double_t, RECONFIGURE_RUNNING,
        "Hue", 0.5, 0., 1.)

gen.add("gain", double_t, RECONFIGURE_RUNNING,
        "Gain", 0., 0., 1.)

gen.add("exposure", double_t, RECONFIGURE_RUNNING,
        "Exposure", 0.5, 0., 1.)

gen.add("frame_id", str_t, RECONFIGURE_CLOSE,
        "ROS tf frame of reference, resolved with tf_prefix unless absolute.",
        "camera")

gen.add("camera_info_url_left", str_t, RECONFIGURE_RUNNING,
        "Left camera [[camera_info_manager#URL_Names|calibration URL]] for this video_mode (uncalibrated if null).", "")

gen.add("camera_info_url_right", str_t, RECONFIGURE_RUNNING,
        "Right camera [[camera_info_manager#URL_Names|calibration URL]] for this video_mode (uncalibrated if null).", "")

exit(gen.generate(PACKAGE, "CameraStereo2", "CameraStereo2"))

